### PR TITLE
Moved url escape from request handlers into updateQueryParam function

### PR DESCRIPTION
**Pull request by RachelTucker**

### CODE REVIEW

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/utils/SafeStringManipulation.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    private SafeStringManipulation() {
+        //pass
+    }
+
+    public static <T> String safeUrlEscape(final T obj) {
+        if (obj == null) {
+            return null;
+        }
+        return UrlEscapers.urlFragmentEscaper().escape(safeToString(obj)).replace("+", "%2B");
```
Comment (rpmoore) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/utils/SafeStringManipulation.java]: Won't this still throw an exception if null is passed to the escape method?

Comment (RachelTucker) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/utils/SafeStringManipulation.java]: Yes, which is why the object is tested for nullness above

Comment (rpmoore) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/utils/SafeStringManipulation.java]: Oh, I see it.  I missed it.  Are there some unit tests for this?

Comment (RachelTucker) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/utils/SafeStringManipulation.java]: Yes, I created unit tests for both functions in the SafeStringManipulation class

**Pull request merged**